observatory-platform/observatory/platform/terraform_api.py

```diff
@@ -109,82 +209,83 @@ def workspace_id(self, organisation: str, workspace: str) -> str:
             logging.debug(f"response: {response.text}")
         else:
             logging.error(f"Response status: {response.status_code}")
-            logging.error(f"Unsuccessful retrieving workspace id for workspace '{workspace}', response: {response.text}")
+            logging.error(
+                f"Unsuccessful retrieving workspace id for workspace '{workspace}', response: {response.text}")
             exit(os.EX_CONFIG)
 
         workspace_id = json.loads(response.text)['data']['id']
         return workspace_id
 
-    def add_workspace_variable(self, attributes: dict, workspace_id: str) -> str:
-        """
-        Add a new variable to the workspace. Will return an error if the variable already exists.
-        :param attributes: attributes of the variable
+    def add_workspace_variable(self, variable: TerraformVariable, workspace_id: str) -> str:
```

Comment:
For clarity, in the functions that follow, maybe make named variables for the http response codes instead of referring to their numeric value

Reply:
I've replaced these with HTTPStatus code enums.